### ISSUE TITLE
Exchange address books not shown in Directory/Contact panes

### ISSUE THREAD

**Issue by daoane** [ExchangeCalendar/exchangecalendar#209]
Exchange Calendar 5.0.0-alpha1
Thunderbird 60.2.1
(Ubuntu Linux 18.04LTS)

"Add exchange contact folder" seems to work for both hosted exchange and Office365 (folders discovered after "save" in first dialogue), but neither the added address books nor the addresses are shown in the Directory and Contact panes. However, it seems that the global address list of the hosted exchange account is working properly when composing an email.

**Comment by benken-parasoft** [comment]
I see the same behavior in 5.0.0-alpha1 with Thunderbird 60.3.0.

**Comment by Trim** [comment]
Hello,

Same for me with 5.0.0-alpha2 (of this week) and Thunderbird 60.3.1 on Windows 10 18.09.

I've checked in "about:options" and the address book seems well configured.

So, we are certainly just facing a display issue.

**Comment by Trim** [comment]
Well, I think we have an issue with the interface implementation inside `addressbook/interface/exchangeAbRootDirectory/exchangeAbRootDirectory.js`.

I've found these two documentations, I'll need to first read:
  * https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Address_Book_Interfaces
  * https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Address_Book_Examples

**Comment by Trim** [comment]
Thanks, I didn't found the page to download the SDK before.

Now, I was able to update the sdk with this command:

```
FILE=/home/adrien/code/exchangecalendar/addressbook/interface/exchangeAbDirFactory/exchangeAbDirFactory
# comm-central comes from "hg clone https://hg.mozilla.org/comm-central
python ~/build/xulrunner-sdk/sdk/bin/typelib.py -I ~/code/comm-central/mailnews/addrbook/public -o $FILE.xpt $FILE.idl
```

Unfortunately that hadn't help and, currently, I have no other clue to fix this issue :/

Maybe the future of this addon should be either TbSync plug-in or WebExtensions.
I've seen this weekend, that the Thunderbird team is working on [Web Extension API for mail and calendar](https://bugzilla.mozilla.org/buglist.cgi?short_desc=webext&resolution=---&classification=Client%20Software&query_format=advanced&short_desc_type=allwordssubstr&product=Calendar&product=Thunderbird).

**Comment by kewisch** [comment]
If at all possible, I'd suggest avoiding to build .xpt files. It would be better to just use raw js objects in js code, and only interface to xpcom when absolutely necessary. Use of WebExtension APIs is the future though, see also https://thunderbird-webextensions.readthedocs.io/en/latest/addressBooks.html

**Comment by advancingu** [comment]
@Trim What do you mean by "that [didn't] help"? Do you not see your class modifications in the debugger? For example, after I compiled [this change](https://github.com/ExchangeCalendar/exchangecalendar/pull/202/files#diff-139c1e530dbf5deaabea6acc8cc6dc95R59) I was able to see these new methods wherever a `mivFunctions` instance was referenced.

**Comment by rdegelo** [comment]
I've tried to add multiple times which created multiple entries in "about:options". Is it safe to just delete everything that starts with "extensions.exchangecontacts@"?

**Comment by advancingu** [comment]
Fix merged into v5.0.0-alpha3.
